contracts/payment/CancellablePullPayment.sol

```diff
@@ -0,0 +1,55 @@
+pragma solidity ^0.4.11;
+
+
+import "./PullPayment.sol";
+
+
+/*
+ * @title CancellablePullPayment
+ * 
+ * @dev Inherits from PullPayment and allows the payment sender to cancel
+ * @dev a payment before it's withdrawn by the reciever.
+ * @dev Inherit from this contract and use asyncSend instead of send.
+ */
+
+contract CancellablePullPayment is PullPayment {
+
+  mapping(address => mapping(address => uint )) public paymentsByPayer;
```

Comment:
@DavidKnott I see what you mean, looks like this is something related to the Mock contract being able to make pull payments from multiple accounts. I truly don't know if this is the purpose of PullPayments (remember my extremely short understanding of the technology). The mock contract could control this with an _owner_ modifier. 

Also, the reason you want to do this is to be able to refund the payer address. 
```
if (!payer.send(refund)) {
        throw;
      }

```
Am I right? What if the contract inheriting **PullPayment/CancellablePullPayment** wants to handle this according to a different business logic?  Will this be an over-design?